src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/ProjectTelemetry/ProjectLoadTelemetryReporter.cs

```diff
@@ -0,0 +1,110 @@
+
+            var projectCapabilities = projectFileInfo.ProjectTelemetryMetadata.ProjectCapabilities;
+
+            var hashedReferences = GetHashedReferences(metadataReferences);
+            var fileCounts = GetUniqueHashedFileExtensionsAndCounts(projectFileInfo);
+            var isSdkStyleProject = projectFileInfo.ProjectTelemetryMetadata.IsSdkStyle;
+
+            var projectEvent = new ProjectLoadTelemetryEvent(
+                ProjectId: projectId.HashedValue,
+                SessionId: sessionId.HashedValue,
+                // Matches how O# reports output kind - default is 0 (ConsoleApplication)
+                OutputKind: (int?)outputKind ?? 0,
+                ProjectCapabilities: projectCapabilities.ToArray(),
+                TargetFrameworks: new string[] { targetFramework },
+                References: hashedReferences.Select(h => h.HashedValue).ToArray(),
+                FileExtensions: fileCounts.Keys.Select(k => k.HashedValue).ToArray(),
+                FileCounts: fileCounts.Values.ToArray(),
+                SdkStyleProject: isSdkStyleProject);
+
+            await ReportEventAsync(projectEvent, cancellationToken);
+        }
+        catch (Exception ex)
+        {
+            // Don't fail project loading because we failed to report telemetry.  Just log a warning and move on.
+            logger.LogWarning($"Failed to get project telemetry data: {ex.ToString()}");
+        }
+    }
+
+    private static async Task ReportEventAsync(ProjectLoadTelemetryEvent telemetryEvent, CancellationToken cancellationToken)
+    {
+        var instance = LanguageServerHost.Instance;
+        Contract.ThrowIfNull(instance, nameof(instance));
+        var clientLanguageServerManager = instance.GetRequiredLspService<IClientLanguageServerManager>();
+        await clientLanguageServerManager.SendNotificationAsync("workspace/projectConfiguration", telemetryEvent, cancellationToken);
+    }
+
+    private static ImmutableDictionary<HashedString, int> GetUniqueHashedFileExtensionsAndCounts(ProjectFileInfo projectFileInfo)
+    {
+        // Similar to O#, we report the content files + any non-generated source files.
+        var contentFiles = projectFileInfo.ProjectTelemetryMetadata.ContentFilePaths;
+        var sourceFiles = projectFileInfo.Documents
+            .Concat(projectFileInfo.AdditionalDocuments)
+            .Concat(projectFileInfo.AnalyzerConfigDocuments)
+            .Where(d => !d.IsGenerated)
+            .SelectAsArray(d => d.FilePath);
+        var allFiles = contentFiles.Concat(sourceFiles);
+        var filesCounts = allFiles.GroupBy(file => Path.GetExtension(file)).ToImmutableDictionary(kvp => VsTfmAndFileExtHashingAlgorithm.HashInput(kvp.Key), kvp => kvp.Count());
+        return filesCounts;
+    }
+
+    private static ImmutableArray<HashedString> GetHashedReferences(ImmutableArray<string> metadataReferences)
+    {
+        return metadataReferences.SelectAsArray(VsReferenceHashingAlgorithm.HashInput);
+    }
```

Comment:
We're hashing the full path instead of just the file name or something? I'd imagine that makes this useless.

Reply:
Oops, my mistake.